src/main/java/com/example/ForexApp_v1/controller/TransactionController.java

```diff
@@ -4,77 +4,92 @@
 import com.example.ForexApp_v1.model.TransacDTO;
 import com.example.ForexApp_v1.model.UploadedFile;
 import com.example.ForexApp_v1.service.TransacService;
+import jakarta.validation.ConstraintViolationException;
+import jakarta.validation.Valid;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 
 import java.util.ArrayList;
 import java.util.List;
 
+
 @Controller
+@Validated
 public class TransactionController {
     private final TransacService transacService;
     private List<TransacDTO> transacDTOList = new ArrayList<>();
     public TransactionController(TransacService transacService) {
         this.transacService = transacService;
     }
-
-    //adnotacja aby sprawdzać czy dane nie są puste bądź nullem ---np noEmpty
     @GetMapping("/singleTransaction")
     public String showToSingleTransaction(Model model) {
         try {
             model.addAttribute("transacDTO", new TransacDTO());
             return "singleTransaction";
         }catch (Exception e){
```

Comment:
Can you make the exception less generic?

Reply:
Ok